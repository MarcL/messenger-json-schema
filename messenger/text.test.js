const Ajv = require('ajv');
const textSchema = require('./text');

const ajv = new Ajv({allErrors: true, jsonPointers: true});

describe('Text message', () => {
    let validate;
    
    beforeEach(() => {
        validate = ajv.compile(textSchema);
    });

    test('Valid message', () => {
        const givenMessage = {
            text: 'Valid message'
        };

        validate(givenMessage);
        
        expect(validate.errors).toBeNull();
    });

    describe('Invalid message', () => {
        const invalidScenarios = [
            {
                testMessage: 'Message is not a string',
                givenMessage: {
                    text: 100
                }
            },
            {
                testMessage: 'Message missing text property',
                givenMessage: {}
            },
            {
                testMessage: 'Message contains additional properties',
                givenMessage: {
                    text: 'Message',
                    invalid: 'Invalid property'
                }
            }
        ];

        invalidScenarios.forEach(scenario => {
            const {testMessage, givenMessage} = scenario;

            test(testMessage, () => {        
                validate(givenMessage);

                expect(validate.errors).toMatchSnapshot();
            });
        });

        test('Message length is greater than 2000 characters', () => {
            const message = new Array(2001).fill('a').join('');
            const givenMessage = {
                text: message
            };

            validate(givenMessage);

            expect(validate.errors).toMatchSnapshot();
        });
    });
});
