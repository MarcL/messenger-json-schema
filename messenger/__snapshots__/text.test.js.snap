// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text message Invalid message Message contains additional properties 1`] = `
Array [
  Object {
    "dataPath": "",
    "keyword": "additionalProperties",
    "message": "should NOT have additional properties",
    "params": Object {
      "additionalProperty": "invalid",
    },
    "schemaPath": "#/additionalProperties",
  },
]
`;

exports[`Text message Invalid message Message is not a string 1`] = `
Array [
  Object {
    "dataPath": "/text",
    "keyword": "type",
    "message": "should be string",
    "params": Object {
      "type": "string",
    },
    "schemaPath": "#/properties/text/type",
  },
]
`;

exports[`Text message Invalid message Message length is greater than 2000 characters 1`] = `
Array [
  Object {
    "dataPath": "/text",
    "keyword": "maxLength",
    "message": "should NOT be longer than 2000 characters",
    "params": Object {
      "limit": 2000,
    },
    "schemaPath": "#/properties/text/maxLength",
  },
]
`;

exports[`Text message Invalid message Message missing text property 1`] = `
Array [
  Object {
    "dataPath": "",
    "keyword": "required",
    "message": "should have required property 'text'",
    "params": Object {
      "missingProperty": "text",
    },
    "schemaPath": "#/required",
  },
]
`;
